horreum-web/src/components/CustomTable.tsx

```diff
@@ -0,0 +1,257 @@
+import { useEffect, useState } from "react"
+import { 
+    Column,
+    SortingRule,
+    TableState,
+    UseRowSelectRowProps,
+    UseRowSelectState,
+    UseSortByColumnProps,
+    UseSortByState,
+    useRowSelect,
+    useSortBy,
+    useTable
+} from "react-table"
+import { noop } from "../utils"
+import { Card, CardBody, CardFooter, CardHeader, Flex, FlexItem, OnPerPageSelect, OnSetPage, Pagination, PaginationVariant, Skeleton, Spinner, Title } from "@patternfly/react-core"
+import { Table, Tbody, Td, Th, Thead, Tr } from "@patternfly/react-table"
+import { ThSortType } from "@patternfly/react-table/dist/esm/components/Table/base/types"
+
+type Direction = 'asc' | 'desc' | undefined;
+
+export type StickyProps = {
+    isStickyColumn?: boolean,
+    hasLeftBorder?: boolean
+}
+
+type PaginationProps = {
+    top?: boolean
+    bottom?: boolean
+    count?: number,
+    perPage?: number,
+    page?: number,
+    onSetPage?: OnSetPage
+    onPerPageSelect?: OnPerPageSelect,
+    variant?: 'top' | 'bottom' | PaginationVariant,
+    isCompact?: boolean
+}
+
+type CustomTableProps<D extends object> = {
```

Comment:
Here we define a custom table that is using generics, but the only place we specify a type is in `horreum-web/src/domain/schemas/SchemaList.tsx`

If we are going to use generics to enforce type checking at compile time, I think we should extend the usage to cover all instances of `<CustomTable/>`

Reply:
Yeah make sense, cleaner this way! 
Added on all usages :heavy_check_mark: